core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -0,0 +1,906 @@
+            break;
+
+        case "outputMetrics":
+            return "Output Size / Records";
+            break;
+
+        case "peakExecutionMemory":
+            return "Peak Execution Memory";
+            break;
+
+        case "resultSerializationTime":
+            return "Result Serialization Time";
+            break;
+
+        case "schedulerDelay":
+            return "Scheduler Delay";
+            break;
+
+        case "diskBytesSpilled":
+            return "Shuffle spill (disk)";
+            break;
+
+        case "memoryBytesSpilled":
+            return "Shuffle spill (memory)";
+            break;
+
+        case "shuffleReadMetrics":
+            return "Shuffle Read Size / Records";
+            break;
+
+        case "shuffleWriteMetrics":
+            return "Shuffle Write Size / Records";
+            break;
+
+        case "executorDeserializeTime":
+            return "Task Deserialization Time";
+            break;
+
+        default:
+            return "NA";
+    }
+}
+
+function displayRowsForSummaryMetricsTable(row, type, columnIndex) {
+    switch(row.metric) {
+        case 'Input Size / Records':
+            var str = formatBytes(row.data.bytesRead[columnIndex], type) + " / " +
+              row.data.recordsRead[columnIndex];
+            return str;
+            break;
+
+        case 'Output Size / Records':
+            var str = formatBytes(row.data.bytesWritten[columnIndex], type) + " / " +
+              row.data.recordsWritten[columnIndex];
+            return str;
+            break;
+
+        case 'Shuffle Read Size / Records':
+            var str = formatBytes(row.data.readBytes[columnIndex], type) + " / " +
+              row.data.readRecords[columnIndex];
+            return str;
+            break;
+
+        case 'Shuffle Read Blocked Time':
+            var str = formatDuration(row.data.fetchWaitTime[columnIndex]);
+            return str;
+            break;
+
+        case 'Shuffle Remote Reads':
+            var str = formatBytes(row.data.remoteBytesRead[columnIndex], type);
+            return str;
+            break;
+
+        case 'Shuffle Write Size / Records':
+            var str = formatBytes(row.data.writeBytes[columnIndex], type) + " / " +
+              row.data.writeRecords[columnIndex];
+            return str;
+            break;
+
+        default:
+            return (row.metric == 'Peak Execution Memory' || row.metric == 'Shuffle spill (memory)'
+                    || row.metric == 'Shuffle spill (disk)') ? formatBytes(row.data[columnIndex], type) : (formatDuration(row.data[columnIndex]));
+
+    }
+}
+
+function createDataTableForTaskSummaryMetricsTable(task_summary_metrics_table) {
+    var taskMetricsTable = "#summary-metrics-table";
+    if ($.fn.dataTable.isDataTable(taskMetricsTable)) {
+        taskSummaryMetricsDataTable.clear().draw();
+        taskSummaryMetricsDataTable.rows.add(task_summary_metrics_table).draw();
+    } else {
+        var task_conf = {
```

Comment:
this datatable has sorting. It doesn't make a lot of sense to have it, can we remove?

Reply:
Done